src/array.rs

```diff
@@ -216,7 +216,8 @@ impl<T: ImplicitClone + 'static> IArray<T> {
     pub fn get_mut(&mut self) -> Option<&mut [T]> {
         match self {
             Self::Rc(ref mut rc) => Rc::get_mut(rc),
-            Self::Static(_) | Self::Single(_) => None,
+            Self::Static(_) => None,
+            Self::Single(ref mut a) => Some(a),
```

Comment:
change is not reflected in the doc above?

Reply:
Fixed in f80eb14

Reply:
I think its different semantics from `Rc::get_mut`, since every clone of single item arrays will now be mutable and different from original, so `iarray != iarray.clone()...some mutation via get_mut()...` (`Rc` prevents mutation via `get_mut()` if there is a `clone`), but I guess its fine to do that?

Reply:
Leaving this discussion as unresolved as I think it's interesting for the future

Reply:
> I guess its fine to do that?

no idea xD tbh I wasn't too much in favor of adding mutable API. I was a bit in a doubt about that